src/meta_schedule/profiler.cc

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "./utils.h"
+
+namespace tvm {
+namespace meta_schedule {
+
+/**************** Context Manager ****************/
+
+struct ScopedTimer {
+  std::function<void()> func;
+
+  explicit ScopedTimer(std::function<void()> func) : func(func) {}
+  ~ScopedTimer() {
+    LOG(INFO) << "destructed";
+    func();
+  }
+};
```

Comment:
expose this data structure in the header to make it actually compilable

Reply:
Currently it's compilable with a forward declaration. I found it might cause compilation trouble if I put the definition in header file. Maybe we can keep it as it is?

Reply:
It's not going to cause compilation issue if we do it right, like :

```
struct ScopedTimer; // forward decl

class ProfilerNode {...};

struct ScopedTimer { // impl
}
```

Reply:
Got it. Moved to header file.